lib/main.js

```diff
@@ -240,8 +261,10 @@ class RuleEngine {
                         this.selectors.set(selector, fixset);
                         accepted++;
                     }
-                } catch(e if e instanceof SyntaxError) {
-                    this._maybeRaiseSyntaxError(e, raiseErrors, selstr);
+                } catch (e) {
+                    if (e instanceof SyntaxError) {
+                      this._maybeRaiseSyntaxError(e, raiseErrors, selstr);
+                    }
```

Comment:
How does SM not have the extended syntax? Wild.
Anyway, needs re-raising of non-matched exceptions.